### PR TITLE
add insert!

### PR DESCRIPTION
**Pull request by bkamins**
fixes https://github.com/JuliaData/CategoricalArrays.jl/issues/375

### CODE REVIEW

**Comment by bkamins** [comment]
@nalimilan - I have added `insert!`. I have not bumped version as I assume you handle the release process here. Thank you!

File: test/11_array.jl [hunk 1]
```diff
@@ -232,6 +234,30 @@ using CategoricalArrays: DefaultRefType, leveltype
     @test x.pool.levels == x2.pool.levels
     @test x.pool.invindex == x2.pool.invindex
 
+    insert!(x, 1, "a")
+    @test length(x) == 4
```
Comment (bkamins) [inline test/11_array.jl:238]:
```suggestion
    @test length(x) == 7
```

File: test/11_array.jl [hunk 2]
```diff
@@ -232,6 +234,30 @@ using CategoricalArrays: DefaultRefType, leveltype
     @test x.pool.levels == x2.pool.levels
     @test x.pool.invindex == x2.pool.invindex
 
+    insert!(x, 1, "a")
```
Comment (bkamins) [inline test/11_array.jl:237]:
```suggestion
    x = y
    insert!(x, 1, "a")
```

File: test/13_arraycommon.jl [hunk 3]
```diff
@@ -1803,13 +1803,69 @@ end
...
     cases = (["b", "a", missing], Union{String, Missing}["b", "a", "b"])
     @testset "String, has missing: $(any(ismissing.(a)))" for a in cases
         x = CategoricalVector{Union{String, Missing}}(a, ordered=ordered)
 
         push!(x, missing)
         @test x ≅ [a; missing]
+        insert!(x, 1, missing)
+        @test x ≅ [missing; a; missing]
+        insert!(x, 3, missing)
+        @test x ≅ [missing; a; missing; missing]
```
Comment (bkamins) [inline test/13_arraycommon.jl:1868]:
```suggestion
        insert!(x, 2, missing)
        @test x ≅ [missing; missing; a; missing]
```

File: test/13_arraycommon.jl [hunk 4]
```diff
@@ -1820,6 +1876,10 @@ end
...
```
Comment (bkamins) [inline test/13_arraycommon.jl:1882]:
```suggestion
        insert!(x, 2, missing)
        @test x ≅ [missing; missing; a; missing]
```

File: test/11_array.jl [hunk 5]
```diff
@@ -232,6 +234,31 @@ using CategoricalArrays: DefaultRefType, leveltype
     @test x.pool.levels == x2.pool.levels
     @test x.pool.invindex == x2.pool.invindex
 
+    x = y
```
Comment (nalimilan) [inline test/11_array.jl:237]: Can you add the equivalent code to 12_missingarray.jl? These files should be kept in sync. (Or just leave both files alone as tests in 13_arraycommon.jl may be enough.)

Comment (bkamins) [reply test/11_array.jl:237]: added

File: test/12_missingarray.jl [hunk 6]
```diff
@@ -259,6 +261,31 @@ const ≅ = isequal
...
+                @test levels(x) == ["e", "c", "zz"]
+
+                insert!(x, 7, missing)
+                @test length(x) == 7
+                @test x ≅ ["c", missing, missing, "e", "zz", "c", missing]
+                @test levels(x) == ["e", "c", "zz"]
+
+                @test_throws ArgumentError insert!(x, true, missing)
+                @test_throws ArgumentError insert!(x, 0, missing)
+                @test_throws ArgumentError insert!(x, 100, missing)
```
Comment (bkamins) [inline test/12_missingarray.jl:288]:
```suggestion
                @test_throws BoundsError insert!(x, 0, missing)
                @test_throws BoundsError insert!(x, 100, missing)
```

**Comment by nalimilan** [comment]
Thanks! If you like we can make a minor release, otherwise we can wait a bit more until we have other features to ship.

**Comment by bkamins** [comment]
No rush. For your reference I have opened the PR because users wanted to `insert!` into a `DataFrame`, and if they had columns other than `Vector` it failed.

**Pull request merged**